Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.cpp

```diff
@@ -53,6 +54,28 @@ static void replaceTextInRange(WebCore::LocalFrame& frame, const SimpleRange& ra
     frame.editor().replaceSelectionWithText(replacementText, WebCore::Editor::SelectReplacement::Yes, WebCore::Editor::SmartReplace::No, WebCore::EditAction::InsertReplacement);
 }
 
+static std::optional<std::tuple<Node&, DocumentMarker&>> findReplacementMarkerByUUID(WebCore::Document* document, const WTF::UUID& replacementUUID)
+{
+    Node* targetNode { nullptr };
+    DocumentMarker* targetMarker { nullptr };
```

Comment:
These should be smart pointers.

Reply:
what kind should they be?

Reply:
Node is RefCounted so RefPtr is suitable for stack variables. DocumentMarker is not refcounted though so best we can do is WeakPtr at the moment.